app/assets/javascripts/darkswarm/controllers/order_cycle_controller.js.coffee

```diff
@@ -2,15 +2,6 @@ Darkswarm.controller "OrderCycleCtrl", ($scope, $rootScope, $timeout, OrderCycle
   $scope.order_cycle = OrderCycle.order_cycle
   $scope.OrderCycle = OrderCycle
 
-  # Timeout forces this to be evaluated after everything is loaded
-  # This is a hack. We should probably write our own "popover" directive
-  # That takes an expression instead of a trigger, and binds to that
-  $timeout =>
-    $rootScope.$broadcast 'orderCycleSelected'
-    if !$scope.OrderCycle.selected()
-      $("#order_cycle_id").trigger("openTrigger")
```

Comment:
there's something related to tooltips with this openTrigger, I dont know what is it...
here: /javascripts/darkswarm/darkswarm.js.coffee

Reply:
ok, so lets remove that line in javascripts/darkswarm/darkswarm.js.coffee as well?

Reply:
@Matt-Yorkley The problem with broadcasting 'orderCycleSelected' every time the OrderCycleCtrl is initiated is that this occurs every time a tab is loaded, so the user is always navigated to `/shop`.

So I've pushed an alternate way of ensuring update_filters() is run after a product load, using a watcher on the Product.loading flag.

Alternatively I could leave this timeout and use a different event, (naming is hard) and fire update_filter() from that event:

```
  # Timeout forces this to be evaluated after everything is loaded
  # This is a hack. We should probably write our own "popover" directive
  # That takes an expression instead of a trigger, and binds to that
  $timeout =>
    $rootScope.$broadcast 'orderCycleChangeReady'
```

@luisramos0 I also removed the other redundant tooltipProvider reference you mentioned.

Reply:
Also I'm getting different results with tests locally so will see how this lates commit goes, in theory it should pass as the filters are now updating again.

Reply:
> I've pushed an alternate way of ensuring update_filters()

I think this is moving in the right direction, but in this case the value of `Product.loading` can change many times while the user is scrolling through the product list, whereas we need need to call update_filters() only once, when an order cycle is selected...

Reply:
Yes I was afraid of that, hence the alternate suggestion. My question on update_filters being triggered too often is, can. the filters change as more products are loaded? It seems likely as I assume the filters shown are only relevant to what is loaded so far...?

Nonetheless.
- Update_filters() is called when the order cycle changes, via the existing broadcast.
- The remaining problem is when the page is loaded and an order_cycle is already selected.

I'll look at this next week, there must be some safe/more appropriate hook for calling update_filters, my alternate solution, reinstating the timeout in the unrelated OrderCycleCtrl doesn't feel like the best option - it smells like potential edge case race condition.

Reply:
The filters are per order cycle, so they won't change when more products are loaded for the same OC. 

I'm sure there's a nice way to resolve it :+1: